subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/XCTestIntegrationTest.groovy

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest
+
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.app.SwiftLib
+import org.gradle.nativeplatform.fixtures.app.SwiftXcTestTestApp
+import org.gradle.nativeplatform.fixtures.app.TestElement
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import spock.lang.Ignore
+
+import static org.gradle.nativeplatform.fixtures.app.SourceTestElement.newTestCase
+import static org.gradle.nativeplatform.fixtures.app.SourceTestElement.newTestSuite
+
+@Requires([TestPrecondition.SWIFT_SUPPORT, TestPrecondition.MAC_OS_X])
+class XCTestIntegrationTest extends AbstractInstalledToolChainIntegrationSpec {
+
+    def setup() {
+        settingsFile << "rootProject.name = 'app'"
+        buildFile << """
+apply plugin: 'xctest'
+"""
+    }
+
+    def "test tasks are skipped when no source is available"() {
+        when:
+        succeeds("test")
+
+        then:
+        executed(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+        skipped(":createXcTestBundle", ":xcTest")
+
+    }
+
+    def "test task fail when test cases fail"() {
+        def testApp = new SwiftXcTestTestApp([
+                newTestSuite("FailingTestSuite", [
+                        newTestCase("testFail", TestElement.TestCase.Result.FAIL)
+                ])
+        ])
+
+        given:
+        testApp.writeToProject(testDirectory)
+
+        when:
+        fails("test")
+
+        then:
+        executedAndNotSkipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest")
+        testApp.expectedSummaryOutputPattern.matcher(output).find()
```

Comment:
we  should verify that the html and xml reports are generated and contain the expected stuff (can use `DefaultTestExecutionResult` fixture or one of its helpers to do this)